### DISCUSSION TITLE
Group management in aTox

### DISCUSSION THREAD

**Discussion by ghost** [evilcorpltd/aTox#657] [Ideas]
Hello,

I paste you here the message of the developper of qtox:

> For some background:
> 
> The Tox protocol has two types of groups, "text only" groups and "audio and text" groups.
> 
> qTox and uTox both treat "audio and text" groups the same, as a full text group where by default you're not sending audio and dropping any audio you receive, and then having a button to "join the call" which then starts sending audio and playing audio you receive.
> 
> TRIfA treats an audio group as an audio only group with push-to-talk. I've talked to the maintainer of TRIfA and he believes qTox should add the option to create text only groups to work with TRIfA, like uTox has the option for, but qTox only has the option to create audio and text groups now. I think generally having groups where you can do either and default to only text is more versatile and easy to use than having groups that only do one or the other, so I don't really see the value in complicating qTox's group creation, and to me this seem like an UI limitation in TRIfA, but we didn't come to an agreement.
> 
> Note that qTox does support joining both kinds of groups, so will work properly when invited to either kind.
> 
> It looks like TRIfA can create both "text only" and "audio and text" groups in the UI, but , but when trying that myself it's not joining the group itself or sending the invite 😕.
> 
> So it sounds like your issue of "qTox groups are only audio with TRIfA" should really be "Tox audio groups are audio only with TRIfA", since this would happen with any audio Tox group, whether it's created by uTox or qTox.
> 
> The only part of this that's within qTox's control is the ability to create text-only groups, but that still wouldn't solve your issue of "not groups which have both functionalities". For that TRIfA would need to change their UI.

I see you don't have group management now. It will be nice if you discuss with qtox developers, to make something really compatible with them :)

**Comment by robinlinden** [comment]
Hi! Yes, of course aTox groups will work with qTox! 😁

I speak with several qTox developers regularly, and I know at least some of them use aTox. It would be pretty silly to not have groups that work with the most popular clients.

I have an open [issue](https://github.com/evilcorpltd/aTox/issues/157) tracking group support, but adding groups will require some work in [jvm-toxcore-c](https://github.com/TokTok/jvm-toxcore-c) and [jvm-toxcore-api](https://github.com/TokTok/jvm-toxcore-api), so it's been a lower priority for me than getting audio/video working (almost done!)
